packages/frontend/src/components/common/Dashboard/DashboardRefreshButton.tsx

```diff
@@ -0,0 +1,232 @@
+import { Button, Menu, Text, Tooltip } from '@mantine/core';
+import { IconChevronDown, IconRefresh } from '@tabler/icons-react';
+import { useFeatureFlagEnabled } from 'posthog-js/react';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useDashboardRefresh } from '../../../hooks/dashboard/useDashboardRefresh';
+import useToaster from '../../../hooks/toaster/useToaster';
+import { useDashboardContext } from '../../../providers/DashboardProvider';
+import MantineIcon from '../MantineIcon';
+
+const REFRESH_INTERVAL_OPTIONS = [
+    {
+        value: '0.5',
+        label: '30s',
+    },
+    {
+        value: '2',
+        label: '1m',
+    },
+    {
+        value: '5',
+        label: '5m',
+    },
+    {
+        value: '15',
+        label: '15m',
+    },
+    {
+        value: '30',
+        label: '30m',
+    },
+    {
+        value: '60',
+        label: '1h',
+    },
+    {
+        value: '120',
+        label: '2h',
+    },
+];
+
+const DashboardRefreshButtonWithAutoRefresh = () => {
+    const { showToastSuccess } = useToaster();
+    const intervalIdRef = useRef<NodeJS.Timeout | null>(null);
+    const [isOpen, setIsOpen] = useState(false);
+    const [lastRefreshTime, setLastRefreshTime] = useState<Date | null>(null);
+    const [refreshInterval, setRefreshInterval] = useState<
+        undefined | number
+    >();
+
+    const { isFetching, invalidateDashboardRelatedQueries } =
+        useDashboardRefresh();
+    const { clearCacheAndFetch } = useDashboardContext();
+
+    const isOneAtLeastFetching = isFetching > 0;
+
+    const invalidateAndSetRefreshTime = useCallback(() => {
+        clearCacheAndFetch();
+        invalidateDashboardRelatedQueries();
+        setLastRefreshTime(new Date());
+    }, [clearCacheAndFetch, invalidateDashboardRelatedQueries]);
+
+    useEffect(() => {
```

Comment:
Would the mantine useInterval hook work here? https://v6.mantine.dev/hooks/use-interval/

Reply:
ooohh I like that! Let me check 😉

Reply:
seems to work well!